flink-yarn/src/test/java/org/apache/flink/yarn/YarnFileStageTest.java

```diff
@@ -288,10 +285,10 @@ private static void testCopySingleFileFromLocal(
             FileSystem targetFileSystem,
             Path targetDir,
             String localResourceDirectory,
-            TemporaryFolder temporaryFolder)
+            java.nio.file.Path temporaryFolder)
             throws IOException, InterruptedException, URISyntaxException {
 
-        final File srcDir = temporaryFolder.newFolder();
+        final File srcDir = newFolderIn(temporaryFolder).toFile();
```

Comment:
This just seems unnecessary. I'd add a `@TempDir` parameter to each test that requires it and use it as-is.